src/routes/cache-hooks/cache-hooks.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { ExecutedTransaction } from './entities/executed-transaction.entity';
+import { NewConfirmation } from './entities/new-confirmation.entity';
+import { PendingTransaction } from './entities/pending-transaction.entity';
+import { EventType } from './entities/event-payload.entity';
+import { IBalancesRepository } from '../../domain/balances/balances.repository.interface';
+
+@Injectable()
+export class CacheHooksService {
+  constructor(
+    @Inject(IBalancesRepository)
+    private readonly balancesRepository: IBalancesRepository,
+  ) {}
+
+  async onEvent(
+    chainId: string,
+    event: ExecutedTransaction | NewConfirmation | PendingTransaction,
+  ): Promise<void> {
+    switch (event.type) {
+      case EventType.PENDING_MULTISIG_TRANSACTION:
+        break;
+      case EventType.EXECUTED_MULTISIG_TRANSACTION:
+        await this.balancesRepository.clearLocalBalances(
+          chainId,
+          event.address,
+        );
+        break;
```

Comment:
For each event we call the respective functions to clear the local cache.

This can probably be accomplished with some shared behaviour with type unions or by introducing the concept of `InvalidationScope`. For an initial implementation I think this is ok and also easy to read. We can see how this evolves as more endpoints are added in the future.

Reply:
Great! Related to this, just for the sake of curiosity: I was reviewing the current Rust implementation of this, I assume these events are the only ones which could trigger a cache invalidation, right? So `EXECUTED_MULTISIG_TRANSACTION` would invalidate balances cache, and both `PENDING_MULTISIG_TRANSACTION` and `NEW_CONFIRMATION` would invalidate confirmations (?) cache?

Reply:
> Great! Related to this, just for the sake of curiosity: I was reviewing the current Rust implementation of this, I assume these events are the only ones which could trigger a cache invalidation, right? So `EXECUTED_MULTISIG_TRANSACTION` would invalidate balances cache, and both `PENDING_MULTISIG_TRANSACTION` and `NEW_CONFIRMATION` would invalidate confirmations (?) cache?

This is something that we will need to iteratively work on. The current CGW performs a scan over a patter with either the Safe Transaction Hash. This is incredibly inefficient and one of the main goals for the new caching solution is to optimize this step.

Therefore we are now grouping related entries in Sets (in Redis) and we invalidate those not by scanning a pattern but by being more explicit – this means that for each event, we need to figure what what sets need to be invalidated.